src/components/RightSideDrawer.jsx

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+
+import Brightness4Icon from "@mui/icons-material/Brightness4";
+import Brightness7Icon from "@mui/icons-material/Brightness7";
+import LanguageIcon from "@mui/icons-material/Language";
+import MenuIcon from "@mui/icons-material/Menu";
+import PaletteIcon from "@mui/icons-material/Palette";
+import SettingsIcon from "@mui/icons-material/Settings";
+import { Box, List, ListItem, ListItemIcon, ListItemText, Divider, Select, MenuItem } from "@mui/material";
+import Avatar from "@mui/material/Avatar";
+import Drawer from "@mui/material/Drawer";
+import IconButton from "@mui/material/IconButton";
+
+const RightSideDrawer = () => {
+  const [open, setOpen] = React.useState(false);
+  const [isLoggedIn, setIsLoggedIn] = React.useState(false); // Assuming user login state
+  const [themeMode, setThemeMode] = React.useState("light"); // 'light' or 'dark'
+  const [selectedColor, setSelectedColor] = React.useState("primary"); // 'primary' or 'secondary'
+  const [language, setLanguage] = React.useState("English"); // Default language
+  const languageList = ["English", "Spanish", "French", "German", "Italian"]; // Example language list
+
+  const handleToggleDrawer = () => {
+    setOpen(!open);
+  };
+
+  const handleThemeToggle = () => {
+    setThemeMode(themeMode === "light" ? "dark" : "light");
```

Comment:
we are using themeProvider in main.jsx so please consider that and create a way to update the theme mode accordingly

Reply:
Resolved